File: Non-Constructible Change/NonConstructibleChange.java
import java.util.*;

class NonConstructibleChange {

    public int nonConstructibleChange(int[] coins) {
        // Write your code here.
        Arrays.sort(coins);
        int currentChange = 0;
        for (int coin : coins) {
            if (coin > currentChange + 1) {
                return currentChange + 1;
            }
            currentChange += coin;
        }
        return currentChange + 1;
    }
}

File: 5. Non-Constructible Change/NonConstructibleChange.kt
package com.algoexpert.program

fun nonConstructibleChange(coins: MutableList<Int>): Int {
    // Write your code here.
    coins.sort()
    var currentChange = 0
    for (coin in coins) {
        if (coin > currentChange + 1) {
            return currentChange + 1
        }
        currentChange += coin
    }
    return currentChange + 1
}
